ibis/backends/flink/__init__.py

```diff
@@ -0,0 +1,258 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import pyflink.version
+import sqlglot as sg
+from pyflink.table.types import create_arrow_schema
+
+import ibis.common.exceptions as exc
+import ibis.expr.operations as ops
+import ibis.expr.schema as sch
+from ibis.backends.base import BaseBackend, CanListDatabases
+from ibis.backends.base.sql.ddl import fully_qualified_re, is_fully_qualified
+from ibis.backends.flink.compiler.core import FlinkCompiler
+
+if TYPE_CHECKING:
+    import pandas as pd
+    import pyarrow as pa
+    from pyflink.table import TableEnvironment
+
+    import ibis.expr.types as ir
+
+
+class Backend(BaseBackend, CanListDatabases):
+    name = "flink"
+    compiler = FlinkCompiler
+    supports_temporary_tables = True
+    supports_python_udfs = True
+
+    def do_connect(self, t_env: TableEnvironment) -> None:
```

Comment:
I'm assuming that we're going to pass keyword arguments here or some kind of Environment class that we manage instead of expecting the user to construct a `pyflink.table.TableEnvironment` themselves and passing it in?

Reply:
I actually think this is fine/what was suggested by the team. The `TableEnvironment` is an analog for the `SparkSession` that users are expected to pass in for the PySpark backend.

Reply:
Hmm - I wonder what the design considerations are here? I did take a look at other backends and it seems that the pyspark backend is the only one that lets the user pass an Environment object. Does it have to do with the distributed cluster?

Reply:
I believe it has to do with the way you're wrapping the underlying library (PySpark or PyFlink).